gengokit/generator/gen_test.go

```diff
@@ -436,46 +427,53 @@ func renderService(svc *svcdef.Service, prev string, te *gengokit.TemplateExecut
 		return "", err
 	}
 
-	nextBytes, err = testFormat(nextBytes)
+	nextCode, err := testFormat(string(nextBytes))
 	if err != nil {
 		return "", errors.Wrap(err, "cannot format")
 	}
 
-	nextCode := strings.TrimSpace(string(nextBytes))
+	nextCode = strings.TrimSpace(nextCode)
 
 	return nextCode, nil
 }
 
-func testGenerateResponseFile(templFP string, te *gengokit.TemplateExecutor, prevGenMap map[string]io.Reader) ([]byte, error) {
+func diff(a, b string) string {
+	return gentesthelper.DiffStrings(
+		a,
+		b,
+	)
+}
+
+func testGenerateResponseFile(executor *gengokit.Executor, prev io.Reader, templFP string) (string, error) {
```

Comment:
This function is doing the work of getting the `string` out of the `truss.NamedReadWriter` (note that there is a pull request up right now to remove the `truss.NamedReadWriter` interface.

Also generateResponseFile will try and format the output, but it may fail. Though it always returns code. This is useful for manual debugging because truss will always write code to disk, even if it does not format or compile. 

This function on the other had, will return an error if it cannot format, and this is used in the tests to find which files do not format.